### ISSUE TITLE
Filters on array-type properties do not work

### ISSUE THREAD

**Issue by thearchitector** [weaviate/weaviate-python-client#621]
In the new V4 client, filters do not appear to support properties with array types.

If I have a collection `Question` with a property `answer` of type `text[]`, the V3 API works as I would expect with the following dataset and query:

```json
[
  { "answer": ["The fox jumped over the lazy dog."] },
  { "answer": ["banana", "over the hedge"] }
]
```
``` python
client.query.get("Question", ["answer"]).with_where(
    {"path": ["answer"], "operator": "Like", "valueText": "*over*"}
).do()
```
returns both objects, as expected.

The analogous operation in the V4 client does not work:
```python
questions.query.fetch_objects(
    filters=Filter("answer").like("*over*")
)
```
produces the error
```
weaviate.exceptions.WeaviateQueryException: Query call failed with message extract params: data type
filter cannot use "valueText[]" on type "text[]", use "valueText" instead.
```

This error isn't specific to `text[]`, it happens generically.

**Comment by tsmith023** [comment]
Hi @thearchitector, thanks for finding this bug and submitting such a thorough report!

I think this may in fact be an issue on the server-side so I will look into it and get back to you 😁

**Comment by tsmith023** [comment]
@thearchitector, I believe that this error should be fixed with the latest Weaviate server patch release: `v1.22.6`. Are you able to check this with your above MRE? Cheers!

**Comment by dirkkul** [comment]
fixed in the next beta version with weaviate 1.23

**Comment by thearchitector** [comment]
@tsmith023 sorry, a timely response got away from me here.

@dirkkul thanks for the status!
